joomla/joomla-cms
# [plg_content_loadmodule] Prevent issues with regular expression characters

Pull Request for Issue #21329.

### Summary of Changes

This is an alternative to https://github.com/Rosie-Eagle/joomla-cms/pull/1 and https://github.com/joomla/joomla-cms/pull/21948.

This fixes the issue of module positions and modules with titles containing special regular expression characters not being loaded.

### Testing Instructions
Note, the use of `}` hasn't been fixed.

Copied from #21329 by franz-wohlkoenig:
1. Log in at the backend of joomla
2. Go to menu "Extensions > Modules".
3. Create a "Custom" module.
4. Give field 'title' the following value (without the quotes): "Some math: 1 + 1 = 2"
5. Save the module.
6. Next create a new article (or edit an existing one).
7. Make sure you can later view the article at the fronted of joomla, where visitors normally view the article.
8. Enter load module shortcode with your module type and title, e.g. `{loadmodule mod_custom, Some math: 1 + 1 = 2}`
11. Save the article.
12. View the article at the frontend of joomla.

Notice "{loadmodule mod_custom, Some math: 1 + 1 = 2}" not beeing replaced by the actual rendering of the module you created at steps 3 to 5. The text "{loadmodule mod_custom, Some math: 1 + 1 = 2}" is still there as if it were untouched.


### Documentation Changes Required
No.
